addons/hr_timesheet/tests/test_portal_timesheet.py

```diff
@@ -68,3 +69,70 @@ def test_action_view_subtask_timesheet(self):
                 self.assertEqual(view_id, form_view_id)
             elif view_type == 'kanban':
                 self.assertEqual(view_id, kanban_view_id)
+
+    def test_timesheet_multi_company_access(self):
+        """ Ensure a user with access to multiple companies can view
+          timesheets linked to sales orders across companies.
+        """
+        # Skip if required modules are not installed
+        if (self.env['ir.module.module'].search([('name', '=', 'sale_management')]).state == 'uninstalled'
+            and self.env['ir.module.module'].search([('name', '=', 'website')]).state == 'uninstalled'):
+            self.skipTest("This test won't work if sale and website are not installed")
+
+        company_a = self.env.company
+        company_b = self.env['res.company'].search([('id', '=', 3)])   # BE Company CoA
```

Comment:
hard-coded company id = 3, maybe the database doesn't have a company with id = 3